source/extensions/filters/udp/dns_filter/dns_parser.cc

```diff
@@ -0,0 +1,240 @@
+#include "extensions/filters/udp/dns_filter/dns_parser.h"
+
+#include <iomanip>
+#include <sstream>
+
+#include "envoy/network/address.h"
+
+#include "common/common/empty_string.h"
+#include "common/network/address_impl.h"
+#include "common/network/utility.h"
+
+#include "ares.h"
```

Comment:
Did you use all possible code out of ares? Is there more duplication below that could be done using c-ares?

Reply:
I could use a bunch of the macros but my impression is that it's geared to parsing replies and constructing queries.  We could use the response parsing code since it extracts the questions.

Reply:
I would say we should use as much as possible. Can you try to do that and still put in a TODO about fuzzing somewhere?

Reply:
Discussed offline on slack.